From 721c61b152b8451d68ad9e9099f36893ef691da3 Mon Sep 17 00:00:00 2001
From: Thomas Wendt <thoemy@gmx.net>
Date: Mon, 24 Nov 2014 22:00:03 +0100
Subject: [PATCH] libbinder: Add legacy MemoryDealer constructor for old Nvidia
 OMX adaptor

Fixes the following error:

E/hwcomposer(  158): failed to load /system/lib/libnvcap_video.so: dlopen failed: could not load library "libnvomxadaptor.so" needed by "libnvcap_video.so"; caused by cannot locate symbol "_ZN7android12MemoryDealerC1EjPKc" referenced by "libnvomxadaptor.so"

Change-Id: Ib9b96898875fbe6b508bbcf170575919c9728b82
---
 libs/binder/MemoryDealer.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libs/binder/MemoryDealer.cpp b/libs/binder/MemoryDealer.cpp
index 8739625..878f287 100644
--- a/libs/binder/MemoryDealer.cpp
+++ b/libs/binder/MemoryDealer.cpp
@@ -225,6 +225,13 @@ Allocation::~Allocation()
 
 // ----------------------------------------------------------------------------
 
+#ifdef ADD_LEGACY_MEMORY_DEALER_CONSTRUCTOR_SYMBOL
+extern "C" void _ZN7android12MemoryDealerC2EjPKcj(void* obj, size_t size, const char* name, uint32_t flags);
+extern "C" void _ZN7android12MemoryDealerC1EjPKc(void* obj, size_t size, const char* name) {
+    _ZN7android12MemoryDealerC2EjPKcj(obj, size, name, 0);
+}
+#endif
+
 MemoryDealer::MemoryDealer(size_t size, const char* name, uint32_t flags)
     : mHeap(new MemoryHeapBase(size, flags, name)),
     mAllocator(new SimpleBestFitAllocator(size))
-- 
1.9.1

